sycl/include/sycl/atomic_ref.hpp

```diff
@@ -534,15 +540,15 @@ class atomic_ref_impl<
 #else
 class [[__sycl_detail__::__uses_aspects__(aspect::atomic64)]] atomic_ref_impl<
 #endif
-    T, /*IsAspectAtomic64AttrUsed = */ true, DefaultOrder, DefaultScope,
-    AddressSpace, typename std::enable_if_t<std::is_integral_v<T>>>
-    : public atomic_ref_impl<T, /*IsAspectAtomic64AttrUsed = */ false,
-                             DefaultOrder, DefaultScope, AddressSpace> {
+    T, /*SizeOfT = */ 8, DefaultOrder, DefaultScope, AddressSpace,
+    typename std::enable_if_t<std::is_integral_v<T>>>
+    : public atomic_ref_impl<T, /*SizeOfT = */ 4, DefaultOrder, DefaultScope,
+                             AddressSpace> {
 public:
-  using atomic_ref_impl<T, /*IsAspectAtomic64AttrUsed = */ false, DefaultOrder,
-                        DefaultScope, AddressSpace>::atomic_ref_impl;
-  using atomic_ref_impl<T, /*IsAspectAtomic64AttrUsed = */ false, DefaultOrder,
-                        DefaultScope, AddressSpace>::atomic_ref_impl::operator=;
+  using atomic_ref_impl<T, /*SizeOfT = */ 4, DefaultOrder, DefaultScope,
```

Comment:
I understand the idea, but I'm concerned about the naming here. Such generic `SizeOfT` name may prompt someone to use that template argument later to select the right `__spirv` function (or something else) to call. However, here we are re-using implementation for `SizeOfT = 4` for `SizeOfT = 8` type. If that `SizeOfT` was in use anyhow that could have caused problems.

I think that the clearer approach would be  to have here some kind of `enum` template argument which would indicate the need for an aspect to be attached to a class. And a `constexpr` function to convert `T` into that `enum` value that will be used in `atomic_ref` specializations